internal/daemon/daemon.go

```diff
@@ -40,6 +40,30 @@ import (
 	"github.com/canonical/microcluster/state"
 )
 
+// Args are the data needed to start a MicroCluster daemon.
+type Args struct {
+	Verbose bool
+	Debug   bool
+
+	// Consumers of MicroCluster are required to provide a version to serve at /cluster/1.0.
+	Version string
+
+	PreInitListenAddress string
+	HeartbeatInterval    time.Duration
```

Comment:
Can these have comments too?

`PreInitListenAddress` offers the core API and extension servers with `PreInit=true` over the given address/port prior to initializing the daemon. 

`HeaetbeatInterval` configures how often we attempt to send a heartbeat.